climpred/bootstrap.py

```diff
@@ -1,10 +1,58 @@
 import numpy as np
 import xarray as xr
 
-from .prediction import compute_perfect_model, compute_persistence_pm
+from .prediction import (compute_perfect_model, compute_persistence,
+                         compute_reference)
 from .stats import DPP, xr_varweighted_mean_period
 
 
+def resample_uninit_dple_from_lens(dple, lens):
+    """Resample uninitialized hindcast from historical members."""
+    # find range for bootstrapping
+    first_init = max(lens.time.min().values, dple.init.min().values)
+    last_init = min(lens.time.max().values - dple.lead.size,
+                    dple.init.max().values)
+    dple = dple.sel(init=slice(first_init, last_init))
+
+    uninit_dple = []
+    for init in dple.init.values:
+        random_members = np.random.choice(
+            lens.member.values, dple.member.size)
+        # take random uninitialized members from lens at init forcing
+        # (Goddard allows 5 year forcing range here)
+        uninit_at_one_init_year = lens.sel(
+            time=slice(init + 1, init + dple.lead.size)).sel(
+                member=random_members).rename({'time': 'lead'})
+        uninit_at_one_init_year['lead'] = np.arange(
+            1, 1 + uninit_at_one_init_year.lead.size)
+        uninit_at_one_init_year['member'] = np.arange(1,
+                                                      1 + len(random_members))
+        uninit_dple.append(uninit_at_one_init_year)
+    uninit_dple = xr.concat(uninit_dple, 'init')
+    uninit_dple['init'] = dple.init.values
+    return uninit_dple
+
+
+def _distribution_to_ci(ds, ci_low, ci_high, dim='bootstrap'):
+    """Get confidence intervals from bootstrapped distribution."""
```

Comment:
outsourced because both bootstrap functions need this

Reply:
Similarly, would be good to fill out the docstring here for us, even if it is a private function.